hcl/hcl_stack_test.go

```diff
@@ -204,6 +228,29 @@ func TestHCLParserStack(t *testing.T) {
 				},
 			},
 		},
+		{
+			name: "name is not a string after evaluating function - fails",
+			input: []cfgfile{
+				{
+					filename: "stack.tm",
+					body: `
+						terramate {
+							required_version = ""
+						}
+						stack {
+							name = tm_concat([], "test")
+						}
+					`,
+				},
+			},
+			want: want{
+				errs: []error{
+					errors.E(hcl.ErrTerramateSchema,
+						mkrange("stack.tm", start(6, 30, 99), end(6, 34, 103)),
+					),
+				},
+			},
+		},
```

Comment:
```suggestion
		{
			name: "name is not a string after evaluating function - fails",
			input: []cfgfile{
				{
					filename: "stack.tm",
					body: `
						stack {
							name = tm_concat([], "test")
						}
					`,
				},
			},
			want: want{
				errs: []error{
					errors.E(hcl.ErrTerramateSchema,
						mkrange("stack.tm", start(6, 30, 99), end(6, 34, 103)),
					),
				},
			},
		},
```

The `terramate` block is unrelated to the test.It will break the error change..sorry :laughing: (or disregard too)